Example/Auth/Tests/FIRAuthURLPresenterTests.m

```diff
@@ -70,70 +70,72 @@ - (void)testFIRAuthURLPresenterUsingDefaultUIDelegate:(BOOL)usesDefaultUIDelegat
+      SFSafariViewController *viewController = presentViewController;
       XCTAssertTrue([viewController isKindOfClass:[SFSafariViewController class]]);
-    });
-  } else {
-    id mockUIApplicationClass = OCMClassMock([UIApplication class]);
-    OCMStub(ClassMethod([mockUIApplicationClass sharedApplication])).andReturn(mockUIApplication);
-    OCMExpect([mockUIApplication openURL:OCMOCK_ANY]).andDo(^(NSInvocation *invocation) {
-      __unsafe_unretained id unretainedArgument;
-      // Indices 0 and 1 indicate the hidden arguments self and _cmd.
-      // `openURL` is at index 2.
-      [invocation getArgument:&unretainedArgument atIndex:2];
-      XCTAssertEqualObjects(presenterURL, unretainedArgument);
-    });
-  }
+      XCTAssertEqual(viewController.delegate, presenter);
+    } else {
+      UINavigationController *navigationController = presentViewController;
+      XCTAssertTrue([navigationController isKindOfClass:[UINavigationController class]]);
+      FIRAuthWebViewController *webViewController =
+          navigationController.viewControllers.firstObject;
+      XCTAssertTrue([webViewController isKindOfClass:[FIRAuthWebViewController class]]);
+    }
+    [UIPresentationExpectation fulfill];
+  });
 
   // Present the content.
   [presenter presentURL:presenterURL
              UIDelegate:usesDefaultUIDelegate ? nil : mockUIDelegate
         callbackMatcher:callbackMatcher
              completion:completionBlock];
+  [self waitForExpectationsWithTimeout:kExpectationTimeout handler:nil];
   OCMVerifyAll(mockUIDelegate);
-  OCMVerifyAll(mockUIApplication);
-  if ([SFSafariViewController class]) {
-    OCMExpect([mockUIDelegate dismissViewControllerAnimated:OCMOCK_ANY
-                                                 completion:OCMOCK_ANY])
-        .andDo(^(NSInvocation *invocation) {
-      __unsafe_unretained id unretainedArgument;
-      // Indices 0 and 1 indicate the hidden arguments self and _cmd.
-      // `completion` is at index 3.
-      [invocation getArgument:&unretainedArgument atIndex:3];
-      void (^finishBlock)() = unretainedArgument;
-      finishBlock();
-    });
-  }
+
+  // Pretend dismissing view controller.
+  OCMExpect([mockUIDelegate dismissViewControllerAnimated:OCMOCK_ANY
+                                               completion:OCMOCK_ANY])
+      .andDo(^(NSInvocation *invocation) {
+    XCTAssertTrue([NSThread isMainThread]);
+    __unsafe_unretained id unretainedArgument;
+    // Indices 0 and 1 indicate the hidden arguments self and _cmd.
+    // `completion` is at index 3.
+    [invocation getArgument:&unretainedArgument atIndex:3];
+    void (^completion)() = unretainedArgument;
+    dispatch_async(dispatch_get_main_queue(), completion);
+  });
 
   // Close the presented content.
+  completionBlockExpectation = [self expectationWithDescription:@"completion callback"];
+  callbackMatcherExpectation = [self expectationWithDescription:@"callbackMatcher callback"];
```

Comment:
I'd move these two lines above the "// Close the presented content" comment.

Reply:
Done.